test/extended/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -0,0 +1,318 @@
+			errTwo := oc.KubeClient().PolicyV1().Evictions(oc.Namespace()).Evict(context.Background(), eviction)
+			o.Expect(errTwo).To(o.HaveOccurred())
+		})
+	})
+})
+
+func newV1Eviction(ns, evictionName string, deleteOption metav1.DeleteOptions) *policyv1.Eviction {
+	return &policyv1.Eviction{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "policy/v1",
+			Kind:       "Eviction",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      evictionName,
+			Namespace: ns,
+		},
+		DeleteOptions: &deleteOption,
+	}
+}
+
+func pdbStatus(pdb *policyv1.PodDisruptionBudget) *policyv1.PodDisruptionBudgetStatus {
+	if pdb == nil {
+		return nil
+	}
+	return &pdb.Status
+}
+
+func waitForPDBStatus(policyClient policyv1clientset.PolicyV1Interface, objMeta metav1.ObjectMeta, timeout time.Duration, expectedStatus policyv1.PodDisruptionBudgetStatus) (*policyv1.PodDisruptionBudget, error) {
+	return waitForPDB(policyClient, objMeta, timeout, func(p *policyv1.PodDisruptionBudget) (bool, error) {
+		return p.Generation == p.Status.ObservedGeneration &&
+				p.Status.DisruptionsAllowed == expectedStatus.DisruptionsAllowed &&
+				p.Status.CurrentHealthy == expectedStatus.CurrentHealthy &&
+				p.Status.DesiredHealthy == expectedStatus.DesiredHealthy &&
+				p.Status.ExpectedPods == expectedStatus.ExpectedPods,
+			nil
+	})
+
+}
+
+func waitForPDB(policyClient policyv1clientset.PolicyV1Interface, objMeta metav1.ObjectMeta, timeout time.Duration, condition func(pdb *policyv1.PodDisruptionBudget) (bool, error)) (*policyv1.PodDisruptionBudget, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	fieldSelector := fields.OneTermEqualSelector("metadata.name", objMeta.Name).String()
+	lw := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (object runtime.Object, e error) {
+			options.FieldSelector = fieldSelector
+			return policyClient.PodDisruptionBudgets(objMeta.Namespace).List(context.Background(), options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (i watch.Interface, e error) {
+			options.FieldSelector = fieldSelector
+			return policyClient.PodDisruptionBudgets(objMeta.Namespace).Watch(context.Background(), options)
+		},
+	}
+
+	event, err := watchtools.UntilWithSync(ctx, lw, &policyv1.PodDisruptionBudget{}, nil, func(event watch.Event) (bool, error) {
+		if event.Type != watch.Modified && (objMeta.ResourceVersion == "" && event.Type != watch.Added) {
+			return true, fmt.Errorf("different kind of event appeared while waiting for PodDisruptionBudget modification: event: %#v", event)
+		}
+		return condition(event.Object.(*policyv1.PodDisruptionBudget))
+	})
+	if err != nil {
+		if event != nil {
+			return event.Object.(*policyv1.PodDisruptionBudget), err
+		}
+		return nil, err
+	}
+	return event.Object.(*policyv1.PodDisruptionBudget), nil
+}
+
+// isTechPreviewNoUpgrade checks if a cluster is a TechPreviewNoUpgrade cluster
+func isTechPreviewNoUpgrade(oc *exutil.CLI) bool {
```

Comment:
This is 3rd copy of this function, the other two are:
- https://github.com/openshift/origin/blob/2d9edcbacc67a6193412769f2e464ce79161391f/test/extended/builds/volumes.go#L335
- https://github.com/openshift/origin/blob/2d9edcbacc67a6193412769f2e464ce79161391f/test/extended/storage/inline.go#L241

Can you open a followup to pull them into a helper function instead?